### DISCUSSION TITLE
Is it possible to send additional variable from `signInWithOtp()` method to email template

### DISCUSSION THREAD

**Discussion by tawanchaiii** [supabase/supabase#27263] [Questions] [answered]
I try to send additional data when execute `signInWithOtp()`, these data must to show on email that sent with otp.
(FYI: data is not depend on user metadata.)

But. on supabase email template has only these variable that we can use.

```
{{ .ConfirmationURL }} : URL to confirm the e-mail address for the new account
{{ .Token }} : The 6-digit numeric email OTP
{{ .TokenHash }} : The hashed token used in the URL
{{ .SiteURL }} : The URL of the site
{{ .Email }} : The user's email address
{{ .Data }} : The user's user_metadata
{{ .RedirectTo }} : The URL of emailRedirectTo passed in options
```

Do you have any solution to send additional data from `signInWithOtp()` to email template ?

**Comment by GaryAustin1** [comment, marked as answer]
You can replace the email with your own if you have a server involved in your project.  You can use https://supabase.com/docs/reference/javascript/auth-admin-generatelink to get the links you need to format your own email for any auth operation and then send it by calling your SMTP service, or an email providers API if they have one.
